request-server.go

```diff
@@ -158,8 +160,7 @@ func (rs *RequestServer) Serve() error {
 
 	err := rs.serveLoop(pktChan)
 
-	close(pktChan) // shuts down sftpServerWorkers
-	wg.Wait()      // wait for all workers to exit
+	wg.Wait() // wait for all workers to exit
```

Comment:
Won't this wait forever now? 

The previous sequence seems more correct, since you usually close the channels so workers can detect end of record stream and exit. The you wait for the exit.

Reply:
The `pktChan` is closed as a `defer` in the `rs.serveLoop(pktChan)`.

Reply:
Ah yes, I overlooked somehow that the function has been factored out. So the changes you did are completely correct in this regard.